src/stravalib/tests/integration/test_client.py

```diff
@@ -885,6 +888,35 @@ def test_get_segment_effort(mock_strava_api, client):
     assert effort.max_heartrate == 170
 
 
+@pytest.mark.parametrize(
+    "params, warning_type, warning_message",
+    [
+        (
+            {"segment_id": 2345, "athlete_id": 12345},
+            DeprecationWarning,
+            "The '{athlete_id}' parameter is unsupported",
+        ),
+        (
+            {"segment_id": 2345, "limit": 10},
+            DeprecationWarning,
+            "The '{limit}' parameter is deprecated",
+        ),
+    ],
+)
+def test_get_segment_efforts_warnings(
+    params, warning_type, warning_message, mock_strava_api, client
+):
+    """Test that if provided with deprecated params, the user receives a
+    warning."""
+
+    mock_strava_api.get(
+        "/segment_efforts", match=[matchers.query_param_matcher(params)]
+    )
+
+    with pytest.warns(warning_type, match=warning_message):
```

Comment:
I'm working in VS code, and this sometimes works. I can sometimes run the test and see the warning and it captures it if i'm just using my debugger. But it doesn't work if I run the entire test suite. 

I'm really confused as to why it's failing, and I would love any guidance on how to make sure it works properly.